### PR TITLE
add cards for ZPrimeToZRToZWW process

### PR DESCRIPTION
**Pull request by xdlyu**
We want to produce an additional signal by replacing wkk with zkk. The cards are similar with the previous ones: https://github.com/cms-sw/genproductions/tree/master/bin/MadGraph5_aMCatNLO/cards/production/pre2017/13TeV/Triboson%20search

### CODE REVIEW

**Comment by sihyunjeon** [comment]
Hello,
I have one generic comment on the process (just asking for my lesson actually).
So here only decay to W or Z bosons are handled with MadGraph, and probably the rest is handled in Pythia right? Is there a reason not to fully decay bosons to final state objects? As far as I know decaying with Pythia would lose spin correlations (let me refer to MG lauchpad [1]).
Perhaps it's because using Pythia is much faster and spin correlations don't make a big difference to your samples? But still, making full decay chain gridpacks don't seem to be that much time-consuming so I wonder what the reason is.

[1] https://answers.launchpad.net/mg5amcnlo/+question/662752

**Comment by sihyunjeon** [comment]
@agrohsje to me it's good to merge

**Comment by xdlyu** [comment]
> Hello,
> I have one generic comment on the process (just asking for my lesson actually).
> So here only decay to W or Z bosons are handled with MadGraph, and probably the rest is handled in Pythia right? Is there a reason not to fully decay bosons to final state objects? As far as I know decaying with Pythia would lose spin correlations (let me refer to MG lauchpad [1]).
> Perhaps it's because using Pythia is much faster and spin correlations don't make a big difference to your samples? But still, making full decay chain gridpacks don't seem to be that much time-consuming so I wonder what the reason is.
> 
> [1] https://answers.launchpad.net/mg5amcnlo/+question/662752

Hi, we want to have inclusive decay Z for our signal.

**Comment by sihyunjeon** [comment]
@xdlyu that can also be done in MadGraph by writing the decay channels, no?

**Comment by xdlyu** [comment]
> @xdlyu that can also be done in MadGraph by writing the decay channels, no?

Do you mean to use madspin? Actually in our analysis, we do not use variables related with spin correlations. So it is ok to be handled in Pythia.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Triboson_ZPrimeToZRToZWW/Triboson_M_R/Triboson_M_R_customizecards.dat [hunk 1]
```diff
@@ -0,0 +1,6 @@
+set param_card kkinputs 5 <MASS>
+set param_card kkinputs 2 6.708000e+00
+set param_card mass 9000025 <MASS_R>
+set param_card mass 9000023 <MASS>
+set param_card DECAY 9000023 1.195600e+01
+set param_card DECAY 9000025 0.31035
```
Comment (agrohsje) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Triboson_ZPrimeToZRToZWW/Triboson_M_R/Triboson_M_R_customizecards.dat:6]: Do you consider the width stable across the mass range you are probing? The values seems very specific.

Comment (xdlyu) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Triboson_ZPrimeToZRToZWW/Triboson_M_R/Triboson_M_R_customizecards.dat]: We use different width for different signal point. I upload file "width.txt" which contains widths of zkk and r for different signal points and they are calculated in the way as the author of the model recommended.

**Comment by agrohsje** [comment]
Thanks so much! Can you extend the script so that it correctly fills the values?

**Comment by xdlyu** [comment]
> Thanks so much! Can you extend the script so that it correctly fills the values?

Hi, you can find the file called "gen_card.py".

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Triboson_ZPrimeToZRToZWW/Triboson_M_R/Triboson_M_R_customizecards.dat [hunk 2]
```diff
@@ -0,0 +1,6 @@
+set param_card kkinputs 5 <MASS>
+set param_card kkinputs 2 6.708000e+00
```
Comment (agrohsje) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Triboson_ZPrimeToZRToZWW/Triboson_M_R/Triboson_M_R_customizecards.dat:2]: Which param is this? kkinputs 2 ? Maybe I misunderstood.

Comment (xdlyu) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Triboson_ZPrimeToZRToZWW/Triboson_M_R/Triboson_M_R_customizecards.dat:2]: It is one parameter in the coupling.

**Comment by xdlyu** [comment]
@agrohsje Hi, are there still some issues remaining?

**Pull request merged**